symbolic-debuginfo/src/breakpad.rs

```diff
@@ -765,15 +763,64 @@ impl<'d> Iterator for BreakpadStackCfiDeltaRecords<'d> {
     }
 }
 
+/// Possible types of data held by a [`BreakpadStackWinRecord`], as listed in
+/// [http://msdn.microsoft.com/en-us/library/bc5207xw%28VS.100%29.aspx]. Breakpad only deals with
+/// types 0 (`FPO`) and 4 (`FrameData`).
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum BreakpadStackWinRecordType {
+    /// Frame pointer omitted; FPO info available.
+    Fpo = 0,
+
+    /// Frame pointer omitted; Frame data info available.
+    FrameData = 4,
+}
+
 /// A [Windows stack frame record], used on x86.
 ///
 /// Example: `STACK WIN 4 2170 14 1 0 0 0 0 0 1 $eip 4 + ^ = $esp $ebp 8 + = $ebp $ebp ^ =`
 ///
 /// [Windows stack frame record]: https://github.com/google/breakpad/blob/master/docs/symbol_files.md#stack-win-records
-#[derive(Clone, Debug, Default, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub struct BreakpadStackWinRecord<'d> {
-    /// Variables and the program string.
-    pub text: &'d str,
+    /// The type of frame data this record holds.
+    pub frame_type: BreakpadStackWinRecordType,
+
+    /// The starting address covered by this record, relative to the module's load address.
+    pub rva: u64,
+
+    /// The number of bytes covered by this record.
+    pub code_size: u64,
+
+    /// The size of the prologue machine code within the record's range in bytes.
+    pub prologue_size: u64,
+
+    /// The size of the epilogue machine code within the record's range in bytes.
+    pub epilogue_size: u64,
+
+    /// The number of bytes this function expects to be passed as arguments.
+    pub parameter_size: u64,
+
+    /// The number of bytes used by this function to save callee-saves registers.
+    pub saved_register_size: u64,
+
+    /// The number of bytes used to save this function's local variables.
+    pub local_size: u64,
+
+    /// The maximum number of bytes pushed on the stack in the frame.
+    pub max_stack_size: u64,
+
+    /// Whether this record contains a program string.
+    pub has_program_string: bool,
```

Comment:
Would be great to have that flag merged with the `program_string` field as `Option<&str>`. It would be great to keep this as close as possible to [pdb::FrameData](https://docs.rs/pdb/0.7.0/pdb/struct.FrameData.html), which this models directly. See also [write_pdb_stackinfo](https://github.com/getsentry/symbolic/blob/e65aae37fead8acc0402df63351844ae40ac27a0/symbolic-minidump/src/cfi.rs#L612).

Reply:
Does that include using the same sizes for the various parameters?

Reply:
Yes, given that these are the original sizes stored in PDBs, there is little point in changing them.